ccl/backupccl: TestBackupRestoreSystemUsers failed

ccl/backupccl.TestBackupRestoreSystemUsers [failed](https://tanzanite.cluster.engflow.com/invocations/default/05c32f57-330c-4aaf-a1bd-8d78047ceb97?testReportRun=1&testReportShard=23&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [c6b012b6d77c8fda71bd9c4dd24570895f32cbbe](https://github.com/cockroachdb/cockroach/commits/c6b012b6d77c8fda71bd9c4dd24570895f32cbbe):


```
=== RUN   TestBackupRestoreSystemUsers
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestBackupRestoreSystemUsers42823656
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testutils.go:290: no Invalid Descriptors
    testutils.go:290: no Invalid Descriptors
    testutils.go:290: no Invalid Descriptors
    testutils.go:290: no Invalid Descriptors
    backup_test.go:10046: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupRestoreSystemUsers42823656
--- FAIL: TestBackupRestoreSystemUsers (187.58s)
=== RUN   TestBackupRestoreSystemUsers/restore-from-backup-with-existing-user
    backup_test.go:10027: query 'SELECT * FROM system.users': expected:
        admin, , true, 2
        app, NULL, false, 101
        app_role, NULL, true, 102
        root, , false, 1
        test, NULL, false, 103
        test_role, NULL, true, 104
        testuser, NULL, false, 100
        
        got:
        admin, , true, 2
        app, NULL, false, 102
        app_role, NULL, true, 103
        root, , false, 1
        test, NULL, false, 104
        test_role, NULL, true, 105
        testuser, NULL, false, 101
        
    --- FAIL: TestBackupRestoreSystemUsers/restore-from-backup-with-existing-user (40.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=23</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreSystemUsers.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38581

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -9961,19 +9961,19 @@ func TestBackupRestoreSystemUsers(t *testing.T) {
 
 		// Role 'app_role' and user 'app' will be added, and 'app' is granted with 'app_role'
 		// User test will remain untouched with no role granted
-		sqlDBRestore.CheckQueryResults(t, "SELECT * FROM system.users", [][]string{
-			{"admin", "", "true", "2"},
-			{"app", "NULL", "false", "101"},
-			{"app_role", "NULL", "true", "102"},
-			{"root", "", "false", "1"},
-			{"test", "NULL", "false", "100"},
-			{"test_role", "NULL", "true", "103"},
+		sqlDBRestore.CheckQueryResults(t, `SELECT username, "isRole" FROM system.users`, [][]string{
+			{"admin", "true"},
+			{"app", "false"},
+			{"app_role", "true"},
+			{"root", "false"},
+			{"test", "false"},
+			{"test_role", "true"},
 		})
-		sqlDBRestore.CheckQueryResults(t, "SELECT * FROM system.role_members", [][]string{
-			{"admin", "app", "false", "2", "101"},
-			{"admin", "root", "true", "2", "1"},
-			{"app_role", "app", "false", "102", "101"},
-			{"app_role", "test_role", "false", "102", "103"},
+		sqlDBRestore.CheckQueryResults(t, `SELECT "role", member, "isAdmin" FROM system.role_members`, [][]string{
+			{"admin", "app", "false"},
+			{"admin", "root", "true"},
+			{"app_role", "app", "false"},
+			{"app_role", "test_role", "false"},
 		})
 		sqlDBRestore.CheckQueryResults(t, "SHOW USERS", [][]string{
 			{"admin", "", "{}"},
@@ -9994,16 +9994,16 @@ func TestBackupRestoreSystemUsers(t *testing.T) {
 	defer cleanupEmptyCluster1()
 	t.Run("restore-from-backup-with-no-system-role-members", func(t *testing.T) {
 		sqlDBRestore1.Exec(t, "RESTORE SYSTEM USERS FROM $1", localFoo+"/3")
-		sqlDBRestore1.CheckQueryResults(t, "SELECT * FROM system.role_members", [][]string{
-			{"admin", "root", "true", "2", "1"},
+		sqlDBRestore1.CheckQueryResults(t, `SELECT "role", member, "isAdmin"  FROM system.role_members`, [][]string{
+			{"admin", "root", "true"},
 		})
-		sqlDBRestore1.CheckQueryResults(t, "SELECT * FROM system.users", [][]string{
-			{"admin", "", "true", "2"},
-			{"app", "NULL", "false", "100"},
-			{"app_role", "NULL", "true", "101"},
-			{"root", "", "false", "1"},
-			{"test", "NULL", "false", "102"},
-			{"test_role", "NULL", "true", "103"},
+		sqlDBRestore1.CheckQueryResults(t, `SELECT username, "isRole" FROM system.users`, [][]string{
+			{"admin", "true"},
+			{"app", "false"},
+			{"app_role", "true"},
+			{"root", "false"},
+			{"test", "false"},
+			{"test_role", "true"},
 		})
 		sqlDBRestore1.CheckQueryResults(t, "SHOW USERS", [][]string{
 			{"admin", "", "{}"},
@@ -10021,17 +10021,17 @@ func TestBackupRestoreSystemUsers(t *testing.T) {
 		// allocated properly in the restore.
 		sqlDBRestore2.Exec(t, "CREATE USER testuser")
 		sqlDBRestore2.Exec(t, "RESTORE SYSTEM USERS FROM $1", localFoo+"/3")
-		sqlDBRestore2.CheckQueryResults(t, "SELECT * FROM system.role_members", [][]string{
-			{"admin", "root", "true", "2", "1"},
+		sqlDBRestore2.CheckQueryResults(t, `SELECT role, member, "isAdmin" FROM system.role_members`, [][]string{
+			{"admin", "root", "true"},
 		})
-		sqlDBRestore2.CheckQueryResults(t, "SELECT * FROM system.users", [][]string{
-			{"admin", "", "true", "2"},
-			{"app", "NULL", "false", "101"},
-			{"app_role", "NULL", "true", "102"},
-			{"root", "", "false", "1"},
-			{"test", "NULL", "false", "103"},
-			{"test_role", "NULL", "true", "104"},
-			{"testuser", "NULL", "false", "100"},
+		sqlDBRestore2.CheckQueryResults(t, `SELECT username, "isRole" FROM system.users`, [][]string{
+			{"admin", "true"},
+			{"app", "false"},
+			{"app_role", "true"},
+			{"root", "false"},
+			{"test", "false"},
+			{"test_role", "true"},
+			{"testuser", "false"},
 		})
 		sqlDBRestore2.CheckQueryResults(t, "SHOW USERS", [][]string{
 			{"admin", "", "{}"},

